### PR TITLE
test: enrich ut for pkg/protocol/header.go

### PR DESCRIPTION
**Pull request by zstone12**
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
提高对 `hertz/pkg/protocol/header` 的单测覆盖率

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: `hertz/pkg/protocol/header` ut coverage increased from 55.73% to 80.4%
zh(optional): `hertz/pkg/protocol/header` 的单测覆盖率从 55.73% 提升到了 80.4%

#### Which issue(s) this PR fixes:

https://github.com/cloudwego/hertz/issues/257

### CODE REVIEW

File: pkg/protocol/header_test.go [hunk 1]
```diff
@@ -93,6 +154,16 @@ func TestRequestHeaderGet(t *testing.T) {
 	}
 }
 
+func TestResponseHeaderGet(t *testing.T) {
+	h := ResponseHeader{}
+	rightVal := "yyy"
+	h.Set("xxx", rightVal)
+	val := h.Get("xxx")
+	if val != rightVal {
+		t.Fatalf("Unexpected %v. Expected %v", val, rightVal)
```
Comment (L2ncE) [inline pkg/protocol/header_test.go:163]: Feel better using `assert.DeepEqual`

Comment (zstone12) [reply pkg/protocol/header_test.go:163]: get it.

Comment (zstone12) [reply pkg/protocol/header_test.go]: done

**Review by welkeyever — APPROVED** [review]
lgtm

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**